File: Code/Components/DanpheEMR.ServerModel/InventoryModels/PurchaseRequestModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DanpheEMR.ServerModel
{
    public class PurchaseRequestModel
    {
        [Key]
        public int PurchaseRequestId { get; set; }
        public int? PRNumber { get; set; } //max+1 logic
        public int? VendorId { get; set; }
        public DateTime RequestDate { get; set; }
        public string RequestStatus { get; set; } // active,withdrawn,cancelled,pending,complete
        public int? VerificationId { get; set; } //always references the latest verification id
        public string Remarks { get; set; }
        public int? CancelledBy { get; set; }
        public DateTime? CancelledOn { get; set; }
        public string CancelRemarks { get; set; }
        public Boolean IsActive { get; set; }
        public Boolean IsPOCreated { get; set; }
        public int CreatedBy { get; set; }
        public DateTime CreatedOn { get; set; }
        public int? ModifiedBy { get; set; }
        public DateTime? ModifiedOn { get; set; }
        public virtual List<PurchaseRequestItemModel> PurchaseRequestItems { get; set; }
        [NotMapped]
        public string RequestedByName { get; set; }
        [NotMapped]
        public string VendorName { get; set; }
        [NotMapped]
        public int MaxVerificationLevel { get; set; }
        [NotMapped]
        public int CurrentVerificationLevelCount { get; set; }
        [NotMapped]
        public int CurrentVerificationLevel { get; set; }
        [NotMapped]
        public List<int> PermissionIdList { get; set; }
        [NotMapped]
        public bool isVerificationAllowed { get; set; }
        [NotMapped]
        public string VerificationStatus { get; set; }
    }
}

File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/inventory/shared/purchase-request.model.ts
import {
  NgForm,
  FormGroup,
  FormControl,
  Validators,
  FormBuilder,
  ReactiveFormsModule
} from '@angular/forms'
import { PurchaseRequestItemModel } from './purchase-request-item.model';
import * as moment from 'moment';
import { ENUM_GRItemCategory } from '../../shared/shared-enums';

export class PurchaseRequestModel {
  public PurchaseRequestId: number = 0;
  public PRNumber: number;
  public VendorId: number;
  public RequestDate: string = moment().format("YYYY-MM-DD");
  public RequestStatus: string; // active,withdrawn,cancelled,pending,complete
  public VerificationId: number;
  public Remarks: number;
  public CancelledBy: number = null;
  public CancelledOn: string = null;
  public CancelRemarks: string = "";
  public IsActive: boolean = true;
  public IsPOCreated: boolean = false;
  public CreatedBy: number = 0;
  public CreatedOn: string = moment().format("YYYY-MM-DD");
  public ModifiedBy: number = null;
  public ModifiedOn: string = null;
  public PurchaseRequestValidator: FormGroup = null;
  public PurchaseRequestItems: Array<PurchaseRequestItemModel> = new Array<PurchaseRequestItemModel>();


  //other properties
  public VendorName: string;
  public CreditPeriod: number;
  public CurrentVerificationLevelCount: number;
  public CurrentVerificationLevel: number;
  public MaxVerificationLevel: number;
  public isVerificationAllowed: boolean;
  public VerificationStatus: string;
  public PRCategory: string = ENUM_GRItemCategory.Consumables;
  public StoreId: number;
  public RequestFromStoreName: string = '';
  PRGroupId: number;
  constructor() {

    var _formBuilder = new FormBuilder();
    this.PurchaseRequestValidator = _formBuilder.group({});
  }

  public IsDirty(fieldName): boolean {
    if (fieldName == undefined)
      return this.PurchaseRequestValidator.dirty;
    else
      return this.PurchaseRequestValidator.controls[fieldName].dirty;
  }


  public IsValid(): boolean { if (this.PurchaseRequestValidator.valid) { return true; } else { return false; } } public IsValidCheck(fieldName, validator): boolean {
    if (fieldName == undefined) {
      return this.PurchaseRequestValidator.valid;
    }
    else
      return !(this.PurchaseRequestValidator.hasError(validator, fieldName));
  }

}
